### ISSUE TITLE
Add an option to ignore SSL errors

### ISSUE THREAD

**Issue by rbino** [astarte-platform/astarte-device-sdk-go#16]
Right now the Go device SDK can't be used with a docker-compose Astarte installation since it fails when using self-signed certificates.

**Comment by jessequinn** [comment]
you could try to get the CA crt and add it to your certificate authority.

**Comment by jessequinn** [comment]
so how i made it work

```
// Create device
	pool := x509.NewCertPool()
	caStr, err := ioutil.ReadFile("astarte/compose/vernemq-certs/rootCA.crt")
	if err != nil {
		_ = fmt.Errorf("read server ca file fail")
		os.Exit(1)
	}
	block, _ := pem.Decode(caStr)
	if block == nil {
		_ = fmt.Errorf("decode ca file fail")
		os.Exit(1)
	}
	if block.Type != "CERTIFICATE" || len(block.Headers) != 0 {
		_ = fmt.Errorf("decode ca block file fail")
		os.Exit(1)
	}
	cert, err := x509.ParseCertificate(block.Bytes)
	if err != nil {
		_ = fmt.Errorf("ParseCertificate ca block file fail")
		os.Exit(1)
	}
	d, err := device.NewDevice(deviceID, deviceRealm, credentialsSecret, apiEndpoint, &http.Client{
		Transport: &http.Transport{
			TLSClientConfig: &tls.Config{
				RootCAs:    pool,
				ServerName: cert.Subject.CommonName,
			},
		}, Timeout: time.Second * 30})
	if err != nil {
		fmt.Println(err.Error())
		os.Exit(1)
	}
```

**Comment by rbino** [comment]
Given now we can ignore SSL errors (after #20 was merged), I think we can close this. @jessequinn if you specifically need support for a custom CA feel free to open a separate issue/PR.
